osi_vehicle.proto

```diff
@@ -0,0 +1,971 @@
+    //
+    optional InterfaceVersion version = 1;
+
+    //
+    // The data timestamp of the simulation environment. The zero time point is
+    // arbitrary but must be identical for all messages.
+    // Recommendation: Zero time point for start point of the simulation.
+    //
+    optional Timestamp timestamp = 2;
+
+    //
+    // The ID of the host vehicle object.
+    //
+    optional Identifier host_vehicle_id = 3;
+
+    //
+    // Deeper going description of one or more vehicle regarding these aspects:
+    // Vehicle-Basics, Vehicle-Kinematics, Vehicle-Powertrain, Vehicle-SteeringWheel, Vehicle-Wheels,
+    // Vehicle-Light-State, Vehicle-Localization, Vehicle-Automated-Driving-Functions.
+    //
+    repeated VehicleClass vehicle = 4;
+}
+
+//
+// \brief The vehicle-class is a deeper description of a vehicle.
+// It consists of different messages categorizing the vehicle in:
+// Vehicle-Basics, Vehicle-Kinematics, Vehicle-Powertrain, Vehicle-SteeringWheel, Vehicle-Wheels,
+// Vehicle-Light-State, Vehicle-Localization, Vehicle-Automated-Driving-Functions.
+// As it is an description of the whole vehicle (focus on cars) it can be used as interface for various reasons, 
+// e.g. to the vehicle-model for kinematics calculation or even for the visualisation in a graphic-engine.
+//
+message VehicleClass
+{
+    //
+    // The basic parameters of the vehicle.
+    //
+    optional VehicleBasics vehicle_basics = 1;
+    
+    //
+    // This is the interface, that describes how the vehicle is moving.
+    // All coordinates and orientations are relative to the global ground truth
+    // coordinate system.
+    //
+    optional VehicleKinematics vehicle_kinematics = 2;
+    
+    //
+    // Interface regarding the powertrain.
+    //
+    optional VehiclePowertrain vehicle_powertrain = 3;
+    
+    //
+    // Interface regarding the SteeringWheel.
+    // The focus here is on the steering wheel.
+    //
+    optional VehicleSteeringWheel vehicle_steering_wheel = 4;
+    
+    //
+    // Interface regarding the wheels.
+    // The focus here is on the physical description of a wheel.
+    //
+    optional VehicleWheels vehicle_wheels = 5;
+    
+    //
+    // Interface regarding the light.
+    // The focus here is on the states of the lights.
+    //
+    optional MovingObject.VehicleClassification.LightState vehicle_light_state = 6;
+    
+    //
+    // Interface regarding the navigation.
+    //
+    optional VehicleLocalization vehicle_localization = 7;
+    
+    //
+    // Interface regarding automated-driving functions.
+    // The focus here is on states and intentions of a function.
+    //
+    repeated VehicleAutomatedDrivingFunction vehicle_automated_driving_function = 8;
+
+    //
+    // \brief The absolut base parameters of the vehicle.
+    //
+    message VehicleBasics
+    {
+        // The id of the car in the simulation. 
+        //
+        optional Identifier id = 1;
```

Comment:
Identifier id is maybe better placed into the root  of the VehicleClass message to mirror MovingObject etc..
That it is nested forced me to add some extra code in my framework ...

BTW: what is the reasoning behind the name "VehicleClass" ?